From 8da8c9b3301428ada74afe4c4c0c90159aeab4b5 Mon Sep 17 00:00:00 2001
From: fu jin <fu.jin@intel.com>
Date: Fri, 7 Nov 2014 12:31:47 -0800
Subject: [PATCH 1/1] set hcd->phy to NULL since it contains error code.

When phy is not found, set hcd->phy to NULL, which is
treated as a valid handle.  This fixes a USB driver
crash.

Change-Id: I06d061d4e47b2f60487a80db379db5f13bff91a7
Signed-off-by: fu jin <fu.jin@intel.com>
---
 drivers/usb/host/xhci-intel-cap.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/host/xhci-intel-cap.c b/drivers/usb/host/xhci-intel-cap.c
index dcde0845af80..87657edcefa1 100644
--- a/drivers/usb/host/xhci-intel-cap.c
+++ b/drivers/usb/host/xhci-intel-cap.c
@@ -50,6 +50,8 @@ int xhci_intel_vendor_cap_init(struct xhci_hcd *xhci)
 		retval = otg_set_host(hcd->phy->otg, &hcd->self);
 		if (retval)
 			usb_put_phy(hcd->phy);
+	} else {
+		hcd->phy = NULL;
 	}
 
 	xhci_dbg(xhci, "capability init done\n");
-- 
1.7.9.5

